thermo_nerf/nerfacto_config/thermal_nerfacto_dataparser.py

```diff
@@ -0,0 +1,457 @@
+from __future__ import annotations
+
+from dataclasses import dataclass, field
+from pathlib import Path
+from typing import Optional, Type
+
+import numpy as np
+import torch
+from nerfstudio.cameras import camera_utils
+from nerfstudio.cameras.cameras import CAMERA_MODEL_TO_TYPE, Cameras, CameraType
+from nerfstudio.data.dataparsers.base_dataparser import DataparserOutputs
+from nerfstudio.data.dataparsers.nerfstudio_dataparser import (
+    Nerfstudio,
+    NerfstudioDataParserConfig,
+)
+from nerfstudio.data.scene_box import SceneBox
+from nerfstudio.data.utils.dataparsers_utils import (
+    get_train_eval_split_all,
+    get_train_eval_split_filename,
+    get_train_eval_split_fraction,
+    get_train_eval_split_interval,
+)
+from nerfstudio.utils.io import load_from_json
+from nerfstudio.utils.rich_utils import CONSOLE
+
+MAX_AUTO_RESOLUTION = 1600
+
+
+@dataclass
+class ThermalNerfactoDataParserConfig(NerfstudioDataParserConfig):
+    _target: Type = field(default_factory=lambda: ThermalNerfacto)
+    """target class to instantiate"""
+    use_thermal_scene: bool = True
+    """whether to use thermal_scene as data (thermal images in thermal folder)"""
+
+
+@dataclass
+class ThermalNerfacto(Nerfstudio):
+    """Nerfstudio DatasetParser"""
+
+    config: ThermalNerfactoDataParserConfig
+    downscale_factor: Optional[int] = None
+
+    def _generate_dataparser_outputs(self, split="train"):
+        assert (
+            self.config.data.exists()
+        ), f"Data directory {self.config.data} does not exist."
+
+        if self.config.data.suffix == ".json":
+            meta = load_from_json(self.config.data)
+            data_dir = self.config.data.parent
+        else:
+            meta = load_from_json(self.config.data / "transforms.json")
+            data_dir = self.config.data
+
+        image_filenames = []
+        mask_filenames = []
+        depth_filenames = []
+        poses = []
+
+        fx_fixed = "fl_x" in meta
+        fy_fixed = "fl_y" in meta
+        cx_fixed = "cx" in meta
+        cy_fixed = "cy" in meta
+        height_fixed = "h" in meta
+        width_fixed = "w" in meta
+        distort_fixed = False
+        for distort_key in ["k1", "k2", "k3", "p1", "p2", "distortion_params"]:
+            if distort_key in meta:
+                distort_fixed = True
+                break
+        fisheye_crop_radius = meta.get("fisheye_crop_radius", None)
+        fx = []
+        fy = []
+        cx = []
+        cy = []
+        height = []
+        width = []
+        distort = []
+
+        # sort the frames by fname
+        if self.config.use_thermal_scene:
+            key = "thermal_file_path"
+        else:
+            key = "file_path"
```

Comment:
Are we making a whole new class because of just those four line?? I think it would be better to:

* Create a script where we convert a thermoScene transforms.json file to "classic" COLMAP file. We should load the "transforms.json" file and change the fey "file_path" for a specified key value before sending that update json file to nerfactor
* Send a patch upstream to nerfstudio to be able to specify yhe key for the path to be used.

I would propose doing the first option because it will help keep Nerfacto simpler and reading and rewritting a json is "cheap".